config/reader.go

```diff
@@ -486,7 +486,6 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 						return newName
 					})
 				}
-				associatedCfgs[subIdx].Attributes = nil
```

Comment:
Was this line causing strange behavior? I've run into this line before and was always curious if it was necessary to nil-out attributes after conversion.

Reply:
I have the same question. Or more generally, I'd appreciate an explanation of this entire `processConfig` function and its intention. I have a bunch of questions that can help drive this. Is this something we can schedule? Maybe for next week when @benjirewis is in town?

I don't think this PR needs to be blocked on my understanding. I do think this entire function needs to document its intention. Additionally, every `if` statement needs to justify the cases they're checking (e.g: why would `resName` be nil? Can `resName` be nil and `remoteName` not be nil?)

Reply:
will do a pass of documentation and also schedule something for `processConfig` (I will do my best but we can discover things together haha)

Reply:
Sounds good, and thanks for scheduling meeting! Until we better understand the function, I'm a little nervous about modifying lines like this that seem to be "unnecessary". Unless you're sure that the nilling of `Attributes` was part of the bug here (and why that was the case), I'd rather not get rid of this line...

Reply:
The nilling of `Attributes` is definitely part of the bug - when we send the config to the module we only convert `Attributes` in both the resource config and associated config. If the rdk can find a map converter for the associated config attributes, the attributes gets niled out (also behavior for resource configs before https://github.com/viamrobotics/rdk/pull/2897). This wasn't an issue for modules back before https://github.com/viamrobotics/rdk/pull/2897 because the custom resource model wouldn't have been in the rdk, and so it gets skipped. But in this case, the rdk does have datamanager, so the associated configs will have it's attributes niled out. 

We could convert `ConvertedAttributes` instead in the associated config, but that feels counter-intuitive and would actually run into RSDK-6022

Reply:
> I do think we can remove this line, since that'd just bring behavior in the server to be more consistent with https://github.com/viamrobotics/rdk/pull/2897 and it fixes this issue

Yep; reading through that context and the previous PR, that sounds good to me. When I said "I've run into this line before", I was actually unknowingly referring to the line (thought this _was_ that line) that @dgottlieb removed in that PR, and I agree with making the behavior consistent here. Thanks for explaining more 🧑‍🔧 .

Reply:
@dgottlieb added some amount of documentation, I'll be adding more using your gist as a starting point (RSDK-6090).